includes/core/classes/class-settings.php

```diff
@@ -352,22 +459,17 @@ public function get_sub_pages(): array {
 						'name'        => __( 'Event Archive Pages', 'gatherpress' ),
 						'description' => __( 'GatherPress allows you to set event archives to pages you have created.', 'gatherpress' ),
 						'options'     => array(
-							'upcoming_events' => array(
+							'post_or_event_date' => array(
 								'labels' => array(
-									'name' => __( 'Upcoming Events', 'gatherpress' ),
+									'name' => __( 'Show post date as Event date', 'gatherpress' ),
 								),
 								'field'  => array(
-									'type'    => 'autocomplete',
-									'options' => array(
-										'type'  => 'page',
-										'label' => __( 'Select Page', 'gatherpress' ),
-										'limit' => 1,
-									),
+									'type' => 'checkbox',
```

Comment:
Let's add this one option to check or not check by default. in this case, check.
```
'options' => array(
    'default' => true,
),
```

Reply:
This should be hooked to too (so when you uncheck or check you see the event or publish date)

Reply:
It is hooked up. I put the screenshot in PR
![image](https://user-images.githubusercontent.com/5721313/209754186-a295ac63-5272-43dc-a67f-4fb5565c8ad0.png)

Reply:
```if ( ! is_a( $post, '\WP_Post' ) || Event::POST_TYPE !== $post->post_type || 1 !== intval( $gp_settings['pages']['post_or_event_date'] ) ) {```

Reply:
Cool, missed that. `if` can be simplified to just 
```
if ( Event::POST_TYPE !== get_post_type( $post ) || 1 !== intval( $gp_settings['pages']['post_or_event_date'] ) ) {
```